app/controllers/api/v1/basic_auth_api/protocols_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module BasicAuthApi
+      class ProtocolsController < BasicAuthApiController
+        def create
+          res = CreateOrUpdateProtocol.run **update_protocol_params
+          return created(res.protocol) if res.valid?
```

Comment:
active interactions don't return what the run function returns, they return an active interaction job. Think of it as a model (it is very similar). From the object you can get access to stuff like errors (so you can do full error management without raising, if you want to), and since it's a PORO, you can make attributes accessible (see line 9 in CreateOrUpdateProtocol)

Reply:
Ah, now that you mention it, I remember that is how it was.